bundles/org.openhab.persistence.influxdb/src/main/java/org/openhab/persistence/influxdb/internal/ItemToStorePointCreator.java

```diff
@@ -64,6 +64,20 @@ public ItemToStorePointCreator(InfluxDBConfiguration configuration, @Nullable Me
     private String calculateMeasurementName(Item item, @Nullable String storeAlias) {
         String name = storeAlias != null && !storeAlias.isBlank() ? storeAlias : item.getName();
 
+        if (configuration.isUseMetadataMeasurementName()) {
+            final MetadataRegistry currentMetadataRegistry = metadataRegistry;
+            if (currentMetadataRegistry != null) {
+                MetadataKey key = new MetadataKey(InfluxDBPersistenceService.SERVICE_NAME, item.getName());
+                Metadata metadata = currentMetadataRegistry.get(key);
+                if (metadata != null) {
+                    String metaName = metadata.getValue();
+                    if (!metaName.isBlank()) {
+                        name = metaName;
+                    }
+                }
+            }
+        }
```

Comment:
This code to search te MetadataRegistry is 3 times the same. Can you put it in a method that used at all these 3 locations?

Reply:
Refactored to a new utils class